typing/typecore.ml

```diff
@@ -2565,6 +2565,7 @@ let collect_apply_args env funct ignore_labels ty_fun ty_fun0 mode_fun sargs =
                      it. *)
                   may_warn funct.exp_loc
                     (Warnings.Non_principal_labels "commuted an argument");
+                  register_allocation_mode mode_fun;
```

Comment:
I think `mode_fun` is the mode of the function that is being closed over, whereas we probably want to register to mode of the closure being created. That is available as `mode_closure` during `type_omitted_parameters`.

Reply:
Thanks, changed.